src/greeter/GreeterApp.cpp

```diff
@@ -134,35 +129,74 @@ namespace SDDM {
 
         m_proxy->setSessionModel(m_sessionModel);
 
+        // create views
+        Q_FOREACH (QScreen *screen, screens())
+            addViewForScreen(screen);
+
+        // handle screens
+        connect(this, &GreeterApp::screenAdded, this, &GreeterApp::addViewForScreen);
+    }
+
+    void GreeterApp::addViewForScreen(QScreen *screen) {
+        // heuristic to detect clone mode, in that case only add a view for the primary screen
+        if (screen->virtualGeometry() == primaryScreen()->geometry() && screen != primaryScreen())
+            return;
+
+        // create view
+        QQuickView *view = new QQuickView();
+        view->setScreen(screen);
+        view->setResizeMode(QQuickView::SizeRootObjectToView);
+        view->setGeometry(QRect(QPoint(0, 0), screen->availableGeometry().size()));
+
+        // remove the view when the screen is removed, but we
+        // need to be careful here since Qt will move the view to
+        // another screen before this signal is emitted so we
+        // pass a pointer to the view to our slot
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 4, 0))
+        connect(this, &GreeterApp::screenRemoved, this, [view, this](QScreen *) {
+            removeViewForScreen(view);
+        });
+#else
+        connect(view, &QQuickView::screenChanged, this, [view, this](QScreen *screen) {
+            if (screen == Q_NULLPTR)
+                removeViewForScreen(view);
+        });
+#endif
+
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 4, 0))
+        // always resize when the screen geometry changes
+        connect(screen, &QScreen::availableGeometryChanged, this, [view](const QRect &r) {
+            view->setGeometry(QRect(QPoint(0, 0), r.size()));
```

Comment:
this isn't right. views geometry is global to the screen (in the X sense of the word) not to the the montitor.
If screen 2 size changes, you'll put it on top of screen 1.

view->setGeometry(screen->geometry()); will do you best